distributed/executor.py

```diff
@@ -193,6 +193,22 @@ def release(self):
             self._cleared = True
             self.executor._dec_ref(tokey(self.key))
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state.update({'event': self.event.is_set(), 'status': self.status})
+        del state['executor']
+        return state
```

Comment:
They're also non-essential.  Generally a rule of thumb is that pickling is a good time to take a hard look at the intrinsic state of an object, removing everything else that is around just for efficiency's sake.